storage/database/db_manager.go

```diff
@@ -244,6 +253,9 @@ type databaseManager struct {
 	config *DBConfig
 	dbs    []Database
 	cm     *cacheManager
+
+	inMigration    bool
```

Comment:
There is no code that makes imMigration false. Is there a concurrency problem in reading and writing this variable?

Reply:
> There is no code that makes imMigration false. 

As the default value of Boolean variable is `false`, only setting it `true` is needed.

> Is there a concurrency problem in reading and writing this variable?

Currently, not. Since `partitionedDatabaseDBManager` and `SetStateTrieMigrationDB` are the only two places where the value is set. As `partitionedDatabaseDBManager` is called during the start of the node and `SetStateTrieMigrationDB` is called after the start of the node, there shouldn't be a concurrency issue here.